File: backend/src/models/MemberPayment.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const MemberPayment = sequelize.define('MemberPayment', {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true
    },
    memberId: {
      type: DataTypes.UUID,
      allowNull: true,
      field: 'member_id',
      references: {
        model: 'members',
        key: 'id'
      }
    },
    memberName: {
      type: DataTypes.STRING,
      allowNull: false,
      field: 'member_name'
    },
    spouseName: {
      type: DataTypes.STRING,
      allowNull: true,
      field: 'spouse_name'
    },
    phone1: {
      type: DataTypes.STRING,
      allowNull: true,
      field: 'phone_1'
    },
    phone2: {
      type: DataTypes.STRING,
      allowNull: true,
      field: 'phone_2'
    },
    year: {
      type: DataTypes.INTEGER,
      allowNull: true
    },
    paymentMethod: {
      type: DataTypes.STRING,
      allowNull: true,
      field: 'payment_method'
    },
    monthlyPayment: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true,
      field: 'monthly_payment'
    },
    totalAmountDue: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true,
      field: 'total_amount_due'
    },
    january: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true
    },
    february: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true
    },
    march: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true
    },
    april: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true
    },
    may: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true
    },
    june: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true
    },
    july: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true
    },
    august: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true
    },
    september: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true
    },
    october: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true
    },
    november: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true
    },
    december: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true
    },
    totalCollected: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true,
      field: 'total_collected'
    },
    balanceDue: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true,
      field: 'balance_due'
    },
    paidUpToDate: {
      type: DataTypes.STRING,
      allowNull: true,
      field: 'paid_up_to_date'
    },
    numberOfHousehold: {
      type: DataTypes.INTEGER,
      allowNull: true,
      field: 'number_of_household'
    }
  }, {
    tableName: 'member_payments_2024',
    timestamps: true,
    createdAt: 'created_at',
    updatedAt: 'updated_at'
  });

  // Define associations
  MemberPayment.associate = (models) => {
    MemberPayment.belongsTo(models.Member, {
      foreignKey: 'memberId',
      as: 'member'
    });
  };

  return MemberPayment;
}; 

File: backendJava/src/main/java/church/abunearegawi/backend/model/MemberPayment.java
package church.abunearegawi.backend.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "member_payments_2024")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MemberPayment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "member_id")
    private UUID memberId;

    @Column(name = "member_name", nullable = false)
    private String memberName;

    @Column(name = "spouse_name")
    private String spouseName;

    @Column(name = "phone_1")
    private String phone1;

    @Column(name = "phone_2")
    private String phone2;

    private Integer year;

    @Column(name = "payment_method")
    private String paymentMethod;

    @Column(name = "monthly_payment", precision = 10, scale = 2)
    private BigDecimal monthlyPayment;

    @Column(name = "total_amount_due", precision = 10, scale = 2)
    private BigDecimal totalAmountDue;

    // Monthly payment columns
    @Column(precision = 10, scale = 2)
    private BigDecimal january;

    @Column(precision = 10, scale = 2)
    private BigDecimal february;

    @Column(precision = 10, scale = 2)
    private BigDecimal march;

    @Column(precision = 10, scale = 2)
    private BigDecimal april;

    @Column(precision = 10, scale = 2)
    private BigDecimal may;

    @Column(precision = 10, scale = 2)
    private BigDecimal june;

    @Column(precision = 10, scale = 2)
    private BigDecimal july;

    @Column(precision = 10, scale = 2)
    private BigDecimal august;

    @Column(precision = 10, scale = 2)
    private BigDecimal september;

    @Column(precision = 10, scale = 2)
    private BigDecimal october;

    @Column(precision = 10, scale = 2)
    private BigDecimal november;

    @Column(precision = 10, scale = 2)
    private BigDecimal december;

    @Column(name = "total_collected", precision = 10, scale = 2)
    private BigDecimal totalCollected;

    @Column(name = "balance_due", precision = 10, scale = 2)
    private BigDecimal balanceDue;

    @Column(name = "paid_up_to_date")
    private String paidUpToDate;

    @Column(name = "number_of_household")
    private Integer numberOfHousehold;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
